api/src/permissions/modules/validate-access/lib/validate-item-access.ts

```diff
@@ -43,22 +58,60 @@ export async function validateItemAccess(options: ValidateItemAccessOptions, con
 		},
 	};
 
+	// Inject the children collection fields after the permissions have been processed, as to not require access to all collection fields
+	if (options.returnAllowedRootFields) {
+		ast.children = Object.keys(context.schema.collections[options.collection]!.fields).map((field) => ({
```

Comment:
As this is after the ast has been processed we should deduplicate the `fields` passed in to ensure no duplicate where conditions are injected. This is assuming `injectCases` does not deduplicate already.

Reply:
For the deduplication, if I handle the case where both fields and returnAllowedRootFields are passed, I can't use injectCases directly since it replaces all cases. I'd need to use getCases to inject cases only for the new fields (with offset-adjusted indices) to preserve the work done by processAst.
That said, this case is pretty rare. Do we really need to handle it?

Reply:
> For the deduplication, if I handle the case where both fields and returnAllowedRootFields are passed, I can't use injectCases directly since it replaces all cases. I'd need to use getCases to inject cases only for the new fields (with offset-adjusted indices) to preserve the work done by processAst.

If it replaces all cases would that not mean it should already be deduplicating the where cases? 

> That said, this case is pretty rare. Do we really need to handle it?

If needed, yes we should.